AllJoyn/Platform/BridgeHost/App.xaml.cs

```diff
@@ -0,0 +1,359 @@
+            await OnboardRegisteredDevices();
+        }
+
+        private bool IsSupportedAppServiceClient(string callerPackageFamilyName)
+        {
+            return callerPackageFamilyName.Equals("org.opent2t.console_h35559jr9hy9m", StringComparison.OrdinalIgnoreCase) || // OpenT2T Console app
+                callerPackageFamilyName.Equals("OpenT2TAllJoynBridgeUWPTestClient_a6049pvczh34w", StringComparison.OrdinalIgnoreCase); // Test app
+        }
+
+        private async void AppServiceConnection_RequestReceived(AppServiceConnection sender, AppServiceRequestReceivedEventArgs args)
+        {
+            AppServiceDeferral msgDeferral = args.GetDeferral();
+            string response = "failed";
+
+            var onboardingData = args.Request.Message;
+            if ((onboardingData != null) && onboardingData.Keys.Contains("command"))
+            {
+                string command = onboardingData["command"] as string;
+                Debug.WriteLine("Command Received: " + command);
+                if (command.Equals("AddDevice"))
+                {
+                    try
+                    {
+                        await AddDeviceToAllJoynAsync(onboardingData);
+                        response = "succeeded";
+                    }
+                    catch (Exception e)
+                    {
+                        response = e.Message;
+                    }
+                }
+                else if ((command != null) && command.Equals("healthcheck"))
+                {
+                    response = "healthy";
+                }
+            }
+            ValueSet valueSet = new ValueSet();
+            valueSet.Add("response", response);
+            await args.Request.SendResponseAsync(valueSet);
+            msgDeferral.Complete();
+        }
+
+        private void TaskInstance_Canceled(IBackgroundTaskInstance sender, BackgroundTaskCancellationReason reason)
+        {
+            appServiceDeferral.Complete();
+            System.Diagnostics.Debug.WriteLine("Task instance cancelled: {0}-{1}", sender.Task.Name, reason.ToString());
+        }
+
+        private  void AppServiceConnection_ServiceClosed(AppServiceConnection sender, AppServiceClosedEventArgs args)
+        {
+            appServiceDeferral.Complete();
+            System.Diagnostics.Debug.WriteLine("service close ConnectionForBackGroundProcess_ServiceClosed");
+        }
+
+        async Task OnboardRegisteredDevices()
+        {
+            // Cache the onboarding data and add the device to the bridge 
+            var appData = Windows.Storage.ApplicationData.Current.LocalSettings;
+
+            // Clear the cache the onboarded devices
+            var onboardedDevicesContainer = appData.CreateContainer(OnboardedDevicesContainerName, ApplicationDataCreateDisposition.Always);
+            onboardedDevicesContainer.Values.Clear();
+
+            foreach (var containerKey in appData.Containers.Keys)
```

Comment:
>Containers [](start = 49, length = 10)

It would be helpful to have some more comments about the way appdata containers are used to store different kinds of data. Especially confusing is the distinction between onboardING data vs onboardED data.